### PR TITLE
Added function to update the dataset status

### PR DESCRIPTION
**Pull request by janvanrijn**
status update fn #528

### CODE REVIEW

File: tests/test_datasets/test_dataset_functions.py [hunk 1]
```diff
@@ -340,3 +340,17 @@ def test_upload_dataset_with_url(self):
...
+        dataset = OpenMLDataset(
+            "UploadTestWithURL", "test", "ARFF",
+            version=1,
+            url="https://www.openml.org/data/download/61/dataset_61_iris.arff")
+        dataset.publish()
+        did = dataset.dataset_id
+
+        openml.datasets.status_update(did, 'active')
+        openml.datasets.status_update(did, 'deactivated')
+        openml.datasets.status_update(did, 'active')
```
Comment (mfeurer) [inline tests/test_datasets/test_dataset_functions.py]: These calls only check if the function succeeds without crashing, but not whether it actually changes something on the server. Could you please check that something actually changed on the server.

File: openml/datasets/functions.py [hunk 2]
```diff
@@ -436,6 +436,31 @@ def create_dataset(name, description, creator, contributor, collection_date,
                          update_comment=update_comment, dataset=arff_dataset)
 
 
+def status_update(data_id, status):
+    """
+    Updates the status of a dataset to either 'active' or 'deactivated'. Please
+    see the OpenML API documentation for a description of the status and all
```
[the remark this answers is not in the document]
Comment (janvanrijn) [reply openml/datasets/functions.py]: Added

File: tests/test_datasets/test_dataset_functions.py (hunk 1 above)
Comment (janvanrijn) [reply tests/test_datasets/test_dataset_functions.py]: Added

File: tests/test_datasets/test_dataset_functions.py [hunk 3]
```diff
@@ -350,8 +350,66 @@ def test_data_status(self):
         dataset.publish()
         did = dataset.dataset_id
 
+        # admin key for test server (only adminds can activate datasets.
+        # all users can deactivate their own datasets)
+        openml.config.apikey = 'd488d8afd93b32331cf6ea9d7003d4c3'
```
[the remark this answers is not in the document]
Comment (janvanrijn) [reply tests/test_datasets/test_dataset_functions.py]: Its only test server, i also put it in the java repo :\ don't know any other solution so it should be fine

File: tests/test_datasets/test_dataset_functions.py [hunk 4]
```diff
@@ -350,8 +350,66 @@ def test_data_status(self):
...
+        self.assertEqual(len(result), 1)
+        self.assertEqual(result[did]['status'], 'active')
         with self.assertRaises(ValueError):
             openml.datasets.status_update(did, 'in_preparation')
+        # need to use listing fn, as this is immune to cache
+        result = openml.datasets.list_datasets(data_id=did, status='all')
+        self.assertEqual(len(result), 1)
+        self.assertEqual(result[did]['status'], 'active')
+
+    def test_create_dataset_numpy(self):
```
[the remark this answers is not in the document]
Comment (janvanrijn) [reply tests/test_datasets/test_dataset_functions.py]: I thought this came up through a merge conflict, thats why i put it in.. Will remove

[the remark this answers is not in the document]
Comment (janvanrijn) [reply tests/test_datasets/test_dataset_functions.py]: Correction, this is not part of this PR. If you look at develop branch, it is already in there. Shall I proceed with merging?

[the remark this answers is not in the document]
Comment (mfeurer) [reply tests/test_datasets/test_dataset_functions.py]: Yes, please go ahead then.

**Pull request merged**